### PR TITLE
Feature/open battle path

### PR DESCRIPTION
**Pull request by Dartemiss**
What's new?
- Roots are removed with camera shake when a battle event is defeated.

PD: When purifying textures are implemented should be done at the same time as this.

### CODE REVIEW

File: Scripts/src/Script/DebugModeScript.cpp [hunk 1]
```diff
@@ -130,6 +130,12 @@ void DebugModeScript::Update()
 	{
 		App->debug_draw->RenderPhysics();
 	}
+
+	if(trigger_roots)
+	{
+		event_manager->ComputeRootPosition();
+		trigger_roots = false;
+	}
```
Comment (JorxSS) [inline Scripts/src/Script/DebugModeScript.cpp:138]: what is the purpose in having this in the Debug Mode?

Comment (Dartemiss) [reply Scripts/src/Script/DebugModeScript.cpp:138]: Not having to defeat all enemies to test it.

File: Scripts/src/Script/EventManager.cpp [hunk 2]
```diff
@@ -63,6 +65,11 @@ void EventManager::Awake()
 	GameObject* audio_music_go = App->scene->GetGameObjectByName("Ambient Music");	
 	audio_source_music = static_cast<ComponentAudioSource*>(audio_music_go->GetComponent(Component::ComponentType::AUDIO_SOURCE));
 
+	GameObject* main_camera_go = App->scene->GetGameObjectByName("Main Camera");
+	const ComponentScript* main_camera_component = main_camera_go->GetComponentScript("CameraShake");
+	camera_shake = (CameraShake*)main_camera_component->script;
```
Comment (JoanStinson) [inline Scripts/src/Script/EventManager.cpp:70]: You can do it in one line of code like this `camera_shake = (CameraShake*)App->scene->GetGameObjectByName("Main Camera")->GetComponentScript("CameraShake")->script;`

File: Scripts/src/Script/EventManager.cpp [hunk 3]
```diff
@@ -278,6 +291,38 @@ bool EventManager::CheckWin()
 	return current_event_beated == 2;
 }
 
+void EventManager::InitRoots()
+{
+	root_obstacles.emplace_back(App->scene->GetGameObjectByName("Roots1"));
+	//root_obstacles.emplace_back(App->scene->GetGameObjectByName("Roots2"));
+	//root_obstacles.emplace_back(App->scene->GetGameObjectByName("Roots3"));
```
Comment (JoanStinson) [inline Scripts/src/Script/EventManager.cpp:298]: Delete comments 💣

File: Scripts/src/Script/EventManager.cpp [hunk 4]
```diff
@@ -278,6 +291,38 @@ bool EventManager::CheckWin()
...
+void EventManager::InitRoots()
+{
+	root_obstacles.emplace_back(App->scene->GetGameObjectByName("Roots1"));
+	//root_obstacles.emplace_back(App->scene->GetGameObjectByName("Roots2"));
+	//root_obstacles.emplace_back(App->scene->GetGameObjectByName("Roots3"));
+}
+
+void EventManager::MoveRoot()
+{
+	camera_shake->trauma = 0.6f;
```
Comment (JoanStinson) [inline Scripts/src/Script/EventManager.cpp:303]: I don't like traumas 😨

File: Scripts/src/Script/EventManager.cpp [hunk 5]
```diff
@@ -278,6 +291,38 @@ bool EventManager::CheckWin()
...
+	
+	float current_lerp = current_time_moving_roots / total_time_moving_roots;
+	float3 new_root_position = float3::Lerp(current_root_position, target_root_position, current_lerp);
+	root_obstacles[current_event]->transform.SetGlobalMatrixTranslation(new_root_position);
+
+	if(current_lerp >= 1.f)
```
Comment (JoanStinson) [inline Scripts/src/Script/EventManager.cpp:309]: ✨ magic

File: Scripts/src/Script/EventManager.cpp [hunk 6]
```diff
@@ -278,6 +291,38 @@ bool EventManager::CheckWin()
...
+		roots_moving = false;
+	}
+
+	current_time_moving_roots += App->time->delta_time;
+}
+
+void EventManager::ComputeRootPosition()
+{
+	current_root_position = root_obstacles[current_event]->transform.GetGlobalTranslation();
+	target_root_position = current_root_position - (float3::unitY * 5.f);
```
Comment (JoanStinson) [inline Scripts/src/Script/EventManager.cpp:321]: magic ✨

File: Scripts/src/Script/EventManager.cpp (hunk 5 above)
Comment (Dartemiss) [reply Scripts/src/Script/EventManager.cpp:309]: It is not magic is an interpolation between 0 and 1!

File: Scripts/src/Script/EventManager.cpp (hunk 2 above)
Comment (Dartemiss) [reply Scripts/src/Script/EventManager.cpp:70]: Done!

File: Scripts/src/Script/EventManager.cpp (hunk 3 above)
Comment (Dartemiss) [reply Scripts/src/Script/EventManager.cpp:298]: Done!

File: Scripts/src/Script/EventManager.cpp (hunk 4 above)
Comment (Dartemiss) [reply Scripts/src/Script/EventManager.cpp:303]: This project is the real trauma 👿

File: Scripts/src/Script/EventManager.cpp (hunk 6 above)
Comment (Dartemiss) [reply Scripts/src/Script/EventManager.cpp:321]: Done!

File: Scripts/src/Script/EventManager.cpp (hunk 5 above)
Comment (JoanStinson) [reply Scripts/src/Script/EventManager.cpp:309]: True!

File: Scripts/src/Script/EventManager.cpp (hunk 3 above)
Comment (JorxSS) [reply Scripts/src/Script/EventManager.cpp]: this could be stay, because at the end in the level we are gonna have those

**Pull request merged**